jobs/looker-utils/looker_utils/unused_dashboards.py

```diff
@@ -0,0 +1,137 @@
+            raise err
+        return("Not Found. Possible Permissions Error")
+        if response.status_code != 500:
+            raise err
+        return ({"items": []}, "skipped")
+    return (response.json(), "completed")
+
+def write_dict_to_csv(json_data, filename):
+    """Write a dictionary to a csv."""
+    with open(filename, "w") as out_file:
+        dict_writer = csv.DictWriter(out_file, CSV_FIELDS)
+        dict_writer.writeheader()
+        dict_writer.writerows(json_data)
+
+def looker_login_post(client_id, client_secret):
+    url = "https://mozilla.cloud.looker.com/api/4.0/login"
+    query_params = {"client_id": client_id, "client_secret": client_secret}
+    response = requests.post(url, query_params)
+    return response
+
+def looker_dashboards_download(access_token):
+    url = "https://mozilla.cloud.looker.com/api/4.0/dashboards"
+    headers = {'Authorization': f'token {access_token}'}
+    params = {}
+    dashboard_data = get_response(url, headers, params)
+    return dashboard_data
+
+
+def looker_one_dashboard_download(submission_date, access_token, dashboard_id):
+    url = f"https://mozilla.cloud.looker.com/api/4.0/dashboards/{dashboard_id}"
+    headers = {'Authorization': f'token {access_token}'}
+    params = {}
+    dashboard_data = {}
+    default_dashboard_dict = {
+        "submission_date": submission_date,
+        "url": f"https://mozilla.cloud.looker.com/{dashboard_id}",
+        "dashboard_id": dashboard_id,
+        "last_accessed_at": None,
+        "last_viewed_at": None,
+        "updated_at": None,
+        "user_name": None,
+        "last_updater_id": None,
+        "last_updater_name": None,
+        "view_count": None,
+    }
+    # this returns a tuple
+    dashboard_data_response = get_response(url, headers, params)
+    if dashboard_data_response == "Not Found. Possible Permissions Error":
+        return default_dashboard_dict
+    dashboard_data = dashboard_data_response[0]
+    last_viewed_str = dashboard_data.get("last_viewed_at")
+    if last_viewed_str:
+        last_viewed_date = datetime.strptime(last_viewed_str, "%Y-%m-%dT%H:%M:%S.%fZ").date()
+        six_months_ago = datetime.today().date() - relativedelta(months=6)
+        if last_viewed_date < six_months_ago:
+            return {
+                **default_dashboard_dict,
+                "last_accessed_at": dashboard_data["last_accessed_at"],
+                "last_viewed_at": dashboard_data["last_viewed_at"],
+                "updated_at": dashboard_data["updated_at"],
+                "user_name": dashboard_data["user_name"],
+                "last_updater_id": dashboard_data["last_updater_id"],
+                "last_updater_name": dashboard_data["last_updater_name"],
+                "view_count": dashboard_data["view_count"],
+            }        
+
+def main():
+    parser = ArgumentParser(description=__doc__)
+    parser.add_argument("--date", required=True)
+
+    args = parser.parse_args()
```

Comment:
might be good to convert this to `click` parameters too, like client_id and client_secret

Reply:
ok. I should probably change it in the `unused_looks` script as well